### ISSUE TITLE
[Bug] Media query and classes rendering issue

### ISSUE THREAD

**Issue by peakrams** [artf/grapesjs#1224]
Hi,

I was trying to make a responsive layout but I discovered that the device switcher does not work consistently. The issue can be replicated in the official demo.

Open http://grapesjs.com/demo.html
Keep the "Desktop" as the selected device.
Add at the top of the "Header" 2 block of type "2 Columns" (just above the menu).
Select the first "Row" inserted and set its padding-left to 100px.
Select the second "Row" inserted and set its padding-left to 100px.
Switch to the device "Tablet".
Add the class "np" to the selected "Row" (switching the device does not deselect it) and change its padding to 10px.
Select the first "Row" and add the class np.

The rendered padding-left is not updated, it is 100px, while I expect it to become 10. NOT good.

Switch the device to "Desktop", the padding-left of both rows is 100px. Good.
Switch the device to "Tablet", the padding-left of both rows is 10px. Good.
Switching multiple times all the devices the padding-left is updated correctly. Good.

Another example.

Open http://grapesjs.com/demo.html
Keep the "Desktop" as the selected device.
Add at the top of the "Header" 2 block of type "2 Columns" (just above the menu).
Select the first "Row" inserted and set its padding-left to 100px.
Select the second "Row" inserted and set its padding-left to 100px.
Switch to the device "Tablet".
Add the class "np" to the selected "Row" (switching the device does not deselect it) and change its padding to 10px.
Select the first "Row" and add the class np.

The rendered padding-left is not updated, it is 100px, while I expect it to become 10.

Switch the device to "Smartphone", the padding-left of both rows is 100px. NOT Good.
Switch the device to "Tablet", the padding-left of both rows is 10px. Good.
Switch the device to "Desktop", the padding-left of both rows is 10px. NOT Good.
Switching multiple time does not solve the issue. NOT Good.

**Comment by peakrams** [comment]
Hello. 

There is or there will be any update about this issue? I've just tried to replicate it in the demo and the issue still persist. Unfortunately, I know only basic JS so I can't think of a way to fix this beaheviour myself. 

I don't know if it is related to my issue but, inspecting the code, in the file config.js the width of the device Tablet is the same as the withMedia of the device Mobile landscape; in this way, the custom style applyed to Mobile landscape via media query won't persist in the canvas switching the device to Tablet? (I didn't try this one, since there is no Mobile landscape in the devices panel.)

Thank you.

**Comment by peakrams** [comment]
Hi, 

can someone help me with this issue? I have just replicated (in the demo) the second example I have provided in the opening post and it seems that nothing has changed from the last time. 

Thank you.

**Comment by artf** [comment]
@peakrams, unfortunately, no one worked on this

**Comment by peakrams** [comment]
Thank you for the reply.

If this can help, I have done a little debug and I have noticed that when you add the class `np` to the second element, the `style` attribute is not updated (this happens both in the model using `editor.getSelected().styleToString()` and in the DOM using the browser inspector) and if you force `editor.getSelected().removeStyle()` the content of the `style` attribute remains the same.

**Comment by peakrams** [comment]
Hi, @arthuralmeidap.

Thank you for your reply. I just did all the steps described in my first post (using Google Chrome on Ubuntu), it's true that something seems to be changed under the hood (at the model level I see no issues) but the view is presented as described in my first post. 

I've just done another test: I added the `np` class and modified its `padding-left` to `10px` before switching the device. Now it seems that the model is updated (I see an empty string if I use the command `editor.getSelected().styleToString()` in the console) but the view is not (in reality, only the first row to which I add the `np` class gets the `10px` as its `padding-left` probably because I set the new `padding-left` when this row is selected, but when I add the `np` class to the other row its `padding-left` doesn't change)... If I try to export the code, in the CSS panel I see the following code 

`.np{
  padding:10px 10px 10px 10px;
}`
...
`@media (max-width: 480px){
  .foot-lists{
    display:none;
  }
}
@media (max-width: 992px){
  .np{
    padding:10px 10px 10px 100px;
  }
}`
